docs/literate/make.jl

```diff
@@ -0,0 +1,122 @@
+using Literate
+using Test
+using HTTP
+import Pkg
+
+# Function to create markdown and notebook files for specific file
+function create_files(title, file; folder="")
+    if folder == ""
+        notebook_filename = "$(splitext(file)[1]).ipynb"
+    else
+        notebook_filename = "$folder/$(splitext(file)[1]).ipynb"
+    end
+
+    binder_badge   = "# [![]($binder_logo)]($binder_url/$notebook_filename)"
+    nbviewer_badge = "# [![]($nbviewer_logo)]($nbviewer_url/$notebook_filename)"
+    download_badge = "# [![]($download_logo)]($download_url/$notebook_filename)"
+    
+    # Generate notebooks
+    function preprocess_notebook(content)
+        return string("# # $title\n\n", preprocess_links(content))
+    end
+    Literate.notebook(joinpath(repo_src, folder, file), joinpath(notebooks_dir, folder); execute=false, preprocess=preprocess_notebook, credit=false)
+
+    # Generate markdowns
+    function preprocess_docs(content)
+        return string("# # [$title](@id $(splitext(file)[1]))\n $binder_badge\n $nbviewer_badge\n $download_badge\n\n", preprocess_links(content))
+    end
+    Literate.markdown(joinpath(repo_src, folder, file), joinpath(pages_dir, folder); preprocess=preprocess_docs,)
+
+    @testset "TrixiTutorials $title" begin include(joinpath(repo_src, folder, file)) end
+
+end
+
+# Creating tutorials for the following files:
+# Normal structure: "title" => "filename.jl"
+# If there are several files for one topic and folder, the structure is:
+#   "title" => ["subtitle 1" => ("folder 1", "filename 1.jl"),
+#               "subtitle 2" => ("folder 2", "filename 2.jl")]
+files = [
+    "Adding a new equation" => ["Scalar conservation law" => ("adding_new_equations_literate", "cubic_conservation_law_literate.jl"),
+                                "Nonconservative equation" => ("adding_new_equations_literate", "nonconservative_advection_literate.jl")],
+    "Differentiable programming" => "differentiable_programming_literate.jl",
+    ]
+
+repo_src        = joinpath(@__DIR__, "src", "files")
+
+pages_dir       = joinpath(@__DIR__, "..", "src", "tutorials")
+notebooks_dir   = joinpath(@__DIR__, "src", "notebooks")
+
+Sys.rm(pages_dir;       recursive=true, force=true)
+Sys.rm(notebooks_dir;   recursive=true, force=true)
+
+# Preprocess files to add reference web links automatically.
+# trixi_version = Pkg.dependencies()[Pkg.project().dependencies["Trixi"]].version
+trixi_link = "https://trixi-framework.github.io/Trixi.jl/v0.3.61/"
```

Comment:
Looks like a TODO note?

Reply:
Since we moved the tutorials back to Trixi.jl, we don't need this `trixi_link` and the following replacing link at all. I will fix this in the next days.